pkg/appgw/internaltypes.go

```diff
@@ -39,6 +40,9 @@ const (
 const (
 	// MaxAllowedHostnames the maximum number of hostnames allowed for listener.
 	MaxAllowedHostnames int = 5
+
+	// WildcardSpecialCharacters are characters that are allowed for wildcard hostnames.
+	WildcardSpecialCharacters = "*?"
```

Comment:
Should we validate this in hostname ?

Reply:
getHostNames function returns a list with the first hostname being the host in the ingress rule. So this should cover it. Is that what you mean ?

Reply:
No I meant, that do we do any validations when user commit the ingress, probably not. I was thinking we should allow users to commit ingress if extendedhostname has any other charactes which we dont support. 
But feel free to ignore as part of this review

Reply:
This is what we will need. https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/